src/main/java/com/flansmod/client/model/RenderGun.java

```diff
@@ -280,14 +281,21 @@ private void renderWeaponSprintMovement(GunAnimations animations, ModelGun model
 			defaultRotation = animations.sprintingStance;
 		}
 
-		GL11.glTranslatef(defaultTranslate.x * progress, defaultTranslate.y * progress, defaultTranslate.z * progress);
-		GL11.glTranslatef(configuredTranslate.x * progress, configuredTranslate.y * progress, configuredTranslate.z * progress);
-		GL11.glRotatef(defaultRotation.x * progress, 1f, 0f, 0f);
-		GL11.glRotatef(defaultRotation.y * progress, 0f, 1f, 0f);
-		GL11.glRotatef(defaultRotation.z * progress, 0f, 0f, 1f);
-		GL11.glRotatef(configuredRotation.x * progress, 1f, 0f, 0f);
-		GL11.glRotatef(configuredRotation.y * progress, 0f, 1f, 0f);
-		GL11.glRotatef(configuredRotation.z * progress, 0f, 0f, 1f);
+		if (!Objects.equals(model.sprintStanceTranslate, new Vector3f(0F, 0F, 0F))) {
```

Comment:
Hm, I think it'd be better to just do the gl calls with (0,0,0) to be honest. Unless that breaks stuff w/ rotation?

Reply:
Not sure what you mean, rotation has to be done this way

Reply:
I mean, how you had it before where it didn't check if the offset was zero, and just applied it anyway

Reply:
I still think this point is valid, Object.equals() feels a bit.. eh. idk

Reply:
Nevermind, I think this is okay actually.